core/modules/user/user.admin.inc

```diff
@@ -168,6 +168,57 @@ function user_admin_settings($form, &$form_state) {
     }
   }
 
+  // Password settings.
+  $form['password_policy'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Password policy'),
+    '#weight' => -1,
+  );
+  $output = '';
+  $criteria = array();
+  $criteria[] = t('not the same as the username');
+  $criteria[] = t('not the same as the e-mail address');
+  $criteria[] = t('sufficiently complex: containing a number of lower- and uppercase letters, digits and symbols');
+  $criteria[] = t('at least as many characters as the minimum number set below');
+
+  $output .= t('The default criteria for a strong password include checking that it is:');
+  $output .= theme('item_list', array('items' => $criteria));
+  $form['password_policy']['user_password_criteria'] = array(
+    '#type' => 'help',
+    '#markup' => $output,
+  );
+  $form['password_policy']['min_length'] = array(
+    '#type' => 'container',
+    '#attributes' => array('class' => array('container-inline min-length-wrapper')),
+  );
+  $form['password_policy']['min_length']['user_password_min_length_enabled'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Password must have a minimum number of characters'),
+    '#title_display' => 'invisible',
+    '#default_value' => $config->get('user_password_min_length_enabled'),
+  );
+  $form['password_policy']['min_length']['user_password_min_length'] = array(
+    '#type' => 'number',
+    '#title' => t('Minimum password length'),
+    '#title_display' => 'invisible',
+    '#min' => 1,
+    '#max' => 64,
+    '#prefix' => t('Password must be at least'),
```

Comment:
Unfortunately, this is not accessible this way ^^, as the prefix of the number field does not tick/untick the checkbox. So let's change that to this then, which works:

```suggestion
    '#title' => t('Password must be at least'),
    '#default_value' => $config->get('user_password_min_length_enabled'),
  );
  $form['password_policy']['min_length']['user_password_min_length'] = array(
    '#type' => 'number',
    '#title' => t('Minimum password length'),
    '#title_display' => 'invisible',
    '#min' => 1,
    '#max' => 64,
```